storagemarket/provider_dealfilter.go

```diff
@@ -54,8 +55,49 @@ func (p *Provider) getDealFilterParams(deal *types.ProviderDealState) (*dealfilt
 
 	return &dealfilter.DealFilterParams{
 		DealParams:           params,
-		SealingPipelineState: *sealingStatus,
+		SealingPipelineState: p.spsCache.Status,
 		FundsState:           *fundsStatus,
 		StorageState:         *storageStatus,
 	}, nil
 }
+
+// sealingPipelineStatus updates the SealingPipelineCache to reduce constant sealingpipeline.GetStatus calls
+// to the lotus-miner. This is to speed up the deal filter processing
+func (p *Provider) sealingPipelineStatus() {
```

Comment:
Because this call is quite expensive, I'd suggest that we don't do in continuously.

Instead I suggest that when there's a call to get the sealing pipeline status
- check the cache
  - if the value has not expired
    - return the value
    - in the background run a process to get a new value
  - if the value has expired
    - get a new value and wait for it before returning

Reply:
I have tweaked the logic a little more. Since we are processing one deal at a time, there is no chance of clash if we don't use goroutines
Filter requests sealing status:

- Check cache
     - if expired or cache error is true 
          - fetch new status (keep the caller waiting)
          - This would also work when Boost starts
          - Return the new status to caller and update cache
     - if cache is not expired
          - Return the cached status to caller (no waiting)
          - Don't trigger a refresh in case we have multiple deals processing at the same time
               - This would prevent the scenario where goroutine and sealingPipelineStatus() both try to update the cache
               - Or we can use locks to prevent this (more complexity)

Reply:
The problem with the code as implemented is that we're repeatedly calling sealingpipeline.GetStatus, even if there are no deal proposals being made. We have seen that this can be a very expensive call (sometimes takes ~30s). So I would prefer to change the logic to only call sealingpipeline.GetStatus when it's needed. That's what I've proposed above.

Reply:
Ahh! Sorry for the confusion in that case.
I have corrected the logic. Pushed new commit with changes.